tests/integration/ipdevpoll/plugins/juniper_alarm_test.py

```diff
@@ -0,0 +1,82 @@
+from mock import Mock
+
+import pytest
+import pytest_twisted
+
+from nav.ipdevpoll.jobs import JobHandler
+from nav.ipdevpoll.plugins import juniperalarm, plugin_registry
+from nav.models.event import EventQueue
+
+
+@pytest.mark.twisted
+@pytest_twisted.inlineCallbacks
+def test_start_event_should_be_posted_on_yellow_and_red_alarms(localhost, db):
+    plugin_registry['juniperalarm'] = juniperalarm.JuniperChassisAlarm
+    job = JobHandler('juniperalarm', localhost.pk, plugins=['juniperalarm'])
+    agent = Mock()
+    job.agent = agent
+    job._create_agentproxy = Mock()
+    job._destroy_agentproxy = Mock()
+    agent.walk.return_value = {
+        '.1.3.6.1.4.1.2636.3.4.2.2.2.0': 1,
+        '.1.3.6.1.4.1.2636.3.4.2.3.2.0': 2,
+    }
+    yield job.run()
+    assert agent.walk.called
+    assert (
+        EventQueue.objects.filter(
+            source_id='ipdevpoll',
+            target_id='eventEngine',
+            event_type='juniperRedAlarmState',
+            netbox_id=localhost.pk,
+            state=EventQueue.STATE_START,
+        ).count()
+        == 1
```

Comment:
I don't entirely agree with this assertion, and we've discussed the reason why several times before:

This does not consider a potentially non-clean database (or the fact that the plugin is in fact stateless, so it *could* post more than one of these events).

IMHO, a more safe assertion might be that the number of matching EventQueue entries is greater than 0, *and also* that the count is *different* than it was before the plugin ran.

Reply:
That's true, I just copied it from `snmpcheck_test`, I will make an issue to also change it there.